packages/pds/src/app-view/services/indexing/plugins/list-item.ts

```diff
@@ -0,0 +1,99 @@
+import { AtUri } from '@atproto/uri'
+import { CID } from 'multiformats/cid'
+import * as ListItem from '../../../../lexicon/types/app/bsky/graph/listitem'
+import * as lex from '../../../../lexicon/lexicons'
+import Database from '../../../../db'
+import {
+  DatabaseSchema,
+  DatabaseSchemaType,
+} from '../../../../db/database-schema'
+import { BackgroundQueue } from '../../../../event-stream/background-queue'
+import RecordProcessor from '../processor'
+
+const lexId = lex.ids.AppBskyGraphListitem
+type IndexedListItem = DatabaseSchemaType['list_item']
+
+const insertFn = async (
+  db: DatabaseSchema,
+  uri: AtUri,
+  cid: CID,
+  obj: ListItem.Record,
+  timestamp: string,
+): Promise<IndexedListItem | null> => {
+  const inserted = await db
+    .insertInto('list_item')
+    .values({
+      uri: uri.toString(),
+      cid: cid.toString(),
+      creator: uri.host,
+      subjectDid: obj.subject,
+      listUri: obj.list,
```

Comment:
I think we'll want to validate that the list item creator is the creator of the list.